addons/project_sale_expense/models/project_project.py

```diff
@@ -16,12 +16,32 @@ def _get_expenses_profitability_items(self, with_action=True):
             groupby=['sale_order_id', 'product_id', 'currency_id'],
             aggregates=['id:array_agg', 'untaxed_amount_currency:sum'],
         )
-        if not expenses_read_group:
-            return {}
         expenses_per_so_id = {}
         expense_ids = []
         dict_amount_per_currency = defaultdict(lambda: 0.0)
-        can_see_expense = with_action and self.env.user.has_group('hr_expense.group_hr_expense_team_approver')
+        can_see_expense = with_action and self.env.user.has_group('hr_expense.group_hr_expense_team_approver') and expenses_read_group
+
+        from_expense = True
+        if not expenses_read_group:
+            from_expense = False
+            expenses_read_group = [
+                (
+                    sol.order_id,
+                    sol.product_id,
+                    sol.currency_id,
+                    [sol.id],
+                    sol.price_subtotal
+                )
+                for sol in self.env['sale.order.line'].sudo().search([
```

Comment:
```suggestion
                for sol in self.env['sale.order.line'].search([
```
here again, is this sudo necessary ?